docs/content/contributing/terraform/terraform-contribution-flow/_index.md

```diff
@@ -0,0 +1,316 @@
+
+```pwsh
+# Create UAMI
+$identityName = "terraform-<provider>-avm-res-<resource provider>-<modulename>" # e.g. terraform-azurerm-avm-res-keyvault-vault
+$resourceGroupName = "<resource group name>"
+$roleName = "Contributor"
+$subscriptionID = $(az account show --query id --output tsv)
+echo "Creating UAMI $identityName, with role $roleName assigned to /subscriptions/$subscriptionID"
+
+$identity=az identity create -g $resourceGroupName -n $identityName
+$principalId=$(az identity show -n $identityName -g $resourceGroupName --query principalId --out tsv)
+az role assignment create --assignee $principalId --role $roleName --scope /subscriptions/$subscriptionID
+
+# Create federated identity credentials
+$fcName = $identityName
+az identity federated-credential create --name $fcName --identity-name $identityName --resource-group $resourceGroupName --issuer "https://token.actions.githubusercontent.com" --subject "repo:Azure/$($identityName):environment:test" --audiences 'api://AzureADTokenExchange'
+```
+
+<br>
+
+---
+
+<br>
+
+### 4. Configure your CI environment
+
+{{< hint type=important >}}
+
+Make sure you have **Docker** installed on your machine.
+
+{{< /hint >}}
+
+1. Download `azterraform` Docker image.
+2. Run `azterraform` Docker image.
+3. ...
+
+<br>
+
+---
+
+<br>
+
+### 5. Implement your contribution
+
+To implement your contribution, we kindly ask you to first review the [shared](/Azure-Verified-Modules/specs/shared/) & [Terraform-specific](/Azure-Verified-Modules/specs/terraform/) specifications and [composition guidelines](/Azure-Verified-Modules/contributing/bicep/terraform/) in particular to make sure your contribution complies with the repository's design and principles.
+
+If you're working on a new module, we'd also ask you to ...
+
+<!--
+TODO:
+- Add implementation sequence in connection with ### 4. Configure your CI environment and ### 6. Run Pre-commit Checks
+-->
+
+<br>
+
+---
+
+<br>
+
+### 6. Run Pre-commit Checks
+
+- [6.1 Run grept](#61-run-grept)
+- [6.2 Check/Generate module documentation](#62-checkgenerate-module-documentation)
+- [6.3 Format Terraform code](#63-format-terraform-code)
+- [6.4 Run e2e tests](#64-run-e2e-tests)
+
+{{< hint type=tip >}}
+
+To simplify and help with the execution of commands like `docscheck`, `terraform-docs`, `terraform-fmt`, etc. there is now a simplified [avm](https://github.com/Azure/terraform-azurerm-avm-template/blob/main/avm) script available in the [`terraform-azurerm-avm-template`](https://github.com/Azure/terraform-azurerm-avm-template) repository which combines all scripts from the [avm_scripts](https://github.com/Azure/tfmod-scaffold/tree/main/avm_scripts) folder in the [tfmod-scaffold](https://github.com/Azure/tfmod-scaffold/) respositroy using [avmmakefile](https://github.com/Azure/tfmod-scaffold/blob/main/avmmakefile). Once you run grept, it will also download/synchronize the avm script for you to your local repository.
+
+{{< /hint >}}
+
+#### 6.1 Run grept
```

Comment:
I would rather we did not run grept manually, we will manage centrally

Reply:
Marking this as `TBD` in our call.

Reply:
moved to owner flow